src/server/BuildableAutomationServer.ts

```diff
@@ -214,13 +214,8 @@ export class BuildableAutomationServer extends AbstractAutomationServer {
     }
 
     private enrichContext(ctx: HandlerContext): HandlerContext {
-        return {
-            teamId: ctx.teamId,
-            correlationId: ctx.correlationId,
-            invocationId: ctx.invocationId,
-            messageClient: ctx.messageClient,
-            graphClient: ctx.graphClient ? ctx.graphClient : this.graphClient,
-        };
+        ctx.graphClient = ctx.graphClient || this.graphClient;
```

Comment:
Before, you were creating a new object, now you are returning the same, possibly modified object. Is this distinction important here?

Reply:
extremely. the context is now mutated and carries information back to the automation.
hmm, which brings up a point. I could put the mutated data in the invoking function, and instead pass a function inside the context which releases it. That would be safer. Then the context would close over some mutable data, but not hold it. Is that better?

Reply:
I don't know.